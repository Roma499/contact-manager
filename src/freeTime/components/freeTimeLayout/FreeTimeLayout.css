.free-time-layout {
  position: relative;
  flex-grow: 1;
}
