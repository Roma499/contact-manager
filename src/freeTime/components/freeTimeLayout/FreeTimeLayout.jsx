import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setLayoutConfig, updateDeal } from '../../actions';
import Deal from '../deal/Deal';
import './FreeTimeLayout.css';

const MINUTES_IN_DAY = 60 * 24;

export class FreeTimeLayout extends Component {
  draggedDealId;
  dealPageYCoord;

  constructor(props) {
    super(props);
    this.layoutRef = React.createRef();
    this.setDraggedDeal = this.setDraggedDeal.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }
  render() {
    return (
      <div
        className="free-time-layout"
        ref={this.layoutRef}
        onDragOver={this.onDragOver}
        onDrop={this.onDrop}
      >
        {this.props.pixelsInMinute &&
          this.props.deals.map(deal => (
            <Deal
              key={deal.id}
              id={deal.id}
              start={deal.start}
              finish={deal.finish}
              setDraggedDeal={this.setDraggedDeal}
            />
          ))}
      </div>
    );
  }
  setDraggedDeal(deal) {
    this.draggedDeal = deal;
  }
  onDragOver(e) {
    e.preventDefault();
    e.dataTransfer.dropEffect = 'move';
  }
  onDrop(e) {
    e.preventDefault();
    this.props.updateDeal(this.draggedDeal);
  }
  componentDidMount() {
    this.props.setLayoutConfig(600 / MINUTES_IN_DAY);
  }
}

function mapStateToProps(state) {
  return {
    pixelsInMinute: state.freeTimeStore.layoutConfig.pixelsInMinute,
    deals: state.freeTimeStore.deals
  };
}
export default connect(
  mapStateToProps,
  { setLayoutConfig, updateDeal }
)(FreeTimeLayout);
