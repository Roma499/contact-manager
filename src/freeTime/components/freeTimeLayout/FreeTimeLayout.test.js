import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { FreeTimeLayout } from './FreeTimeLayout';

function createNodeMock(element) {
  if (element.type === 'div') {
    return {
      current: { clientHeight: 0 }
    };
  }
  return null;
}
function createProps() {
  return {
    pixelsInMinute: null,
    deals: [
      {
        id: 1,
        start: 30600000,
        finish: 31300000
      },
      {
        id: 2,
        start: 40600000,
        finish: 53300000
      }
    ],
    setLayoutConfig: jest.fn(),
    updateDeal: jest.fn()
  };
}

describe('render deal without errors', () => {
  it('renders without crashing', () => {
    mount(<FreeTimeLayout {...createProps()} />);
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(<FreeTimeLayout {...createProps()} />, { createNodeMock })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should invoke once setLayoutConfig ', () => {
    const setLayoutConfig = jest.fn();
    mount(<FreeTimeLayout {...createProps()} setLayoutConfig={setLayoutConfig} />);
    expect(setLayoutConfig.mock.calls.length).toBe(1);
  });
  it('should update invoke once updateDeal onDrop ', () => {
    const updateDeal = jest.fn();
    const wrapper = mount(<FreeTimeLayout {...createProps()} updateDeal={updateDeal} />);
    wrapper.instance().setDraggedDeal({
      id: 2,
      start: 40600000,
      finish: 53300000
    });
    wrapper.find('.free-time-layout').simulate('drop');
    expect(updateDeal.mock.calls.length).toBe(1);
  });
});
