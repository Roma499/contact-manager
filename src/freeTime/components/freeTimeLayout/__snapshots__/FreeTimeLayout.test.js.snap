// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render deal without errors renders correctly 1`] = `
<div
  className="free-time-layout"
  onDragOver={[Function]}
  onDrop={[Function]}
/>
`;
